.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,243 @@
+git remote add upstream https://github.com/dexidp/helm-charts.git
+git fetch upstream
+```
+
+### Making changes
+
+You are ready to make changes to a chart.
+
+Start by creating a new branch:
+
+```bash
+git checkout -b my-branch -t upstream/master
+```
+
+Each pull request should contain changes for a single chart, unless it's a systematic change affecting all or multiple charts.
+
+If you would like to submit breaking changes, please open an issue/discussion first to discuss them with the maintainers.
+
+
+#### Adding variables to `values.yaml`
+
+Changes often involve adding new variables to `values.yaml`.
+If that is the case, please follow the documentation pattern found in `values.yaml` files:
+
+```yaml
+# -- This line documents some variable.
+# Optional line to link to API references or other places that describe this variable in more detail.
+someVariable: 12
+```
+
+Variables often enable/disable certain features based on a default nil value.
+In that case the variable should still be added to the `values.yaml` file so that it can appear in the generated documentation:
+
+```yaml
+# -- (int) This variable has a default nil.
+# Notice the variable type above: it hints the documentation generator to use the correct type.
+someOtherVariable:
+```
+
+For additional `values.yaml` best practices check out the [official Helm documentation](https://helm.sh/docs/chart_best_practices/values/).
+
+
+### Testing
+
+Once you are ready with your changes, make sure they pass the linter:
+
+```bash
+helm lint charts/CHART
+```
+
+Next, you should install the chart to a Kubernetes cluster to make sure your changes don't break existing functionality:
+
+> **Pro tip:** Create a `values.test.yaml` in the chart directory (don't worry about accidentally commiting it: it is in gitignore).
+
+```bash
+helm upgrade --install -f charts/CHART/values.test.yaml my-chart-test charts/CHART
+```
+
+Delete the release from the cluster once you made sure that your changes work:
+
+```bash
+helm delete my-chart-test
+```
+
+For smaller changes (or before installing the chart on a cluster) it might be enough to compile the templates:
+
+```bash
+helm template -f charts/CHART/values.test.yaml my-chart-test charts/CHART
+```
+
+Repeat the test with your feature disabled to make sure the default happy path isn't broken either.
+
+
+### Commiting the changes
+
+Try to keep your changes grouped logically within individual commits.
+It's easier for maintainers to review changes that way.
+
+Once you are done with your changes and testing, commit your changes:
+
+```bash
+git add charts/CHART/
+git commit -m 'feat(charts/CHART): add feature X'
+```
+
+
+#### Commit message guidelines
+
+A good commit message should describe what changed and why.
+This project uses [Conventional Commits](https://www.conventionalcommits.org/) to structure commit messages.
+In the future, commit messages might help automating the release process (eg. automatic chart version bumps).
```

Comment:
Do we need to mention potential future commit message usage? As a future change to the release process would require an edit here, I'd drop it for now and add the info when the feature is implemented.

Reply:
I don't mind dropping it. The reason I mentioned future use cases is because it's hard to reason about using conventional commits if we don't actually make use of it.

Reply:
That's enough of an argument for me if you'd like to keep it in :)